### PR TITLE
Updated gce-pd and redis examples to use v1beta3

### PR DESCRIPTION
**Pull request by ArtfulCoder**
updated gce-pd and redis examples to use v1beta3
#3671

@bgrant0607 
@lavalamp

### CODE REVIEW

File: examples/redis/v1beta3/redis-sentinel-controller.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    spec:
+      containers:
+      - env:
+        - name: SENTINEL
+          value: "true"
+        image: kubernetes/redis:v1
+        name: sentinel
+        ports:
+        - containerPort: 26379
+          name: api
```
Comment (sub-mod) [inline examples/redis/v1beta3/redis-sentinel-controller.yaml]: last i heard we were removing named ports

File: examples/redis/v1beta3/image/redis-master.conf [hunk 2]
```diff
@@ -0,0 +1,827 @@
+# Redis configuration file example
```
Comment (bgrant0607) [inline examples/redis/v1beta3/image/redis-master.conf]: If these .conf files are identical to the v1beta1 versions, I'm inclined to not copy them. Just reuse the existing ones.

File: examples/gce-pd/v1beta3/testpd.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+          # name must match the volume name below
+        - name: testpd
+          # mount path within the container
+          mountPath: /testpd
+  volumes:
+    - name: testpd
+      persistentDisk:
+        # This GCE PD must already exist.
+        pdName: %insert_pd_name_here%
+        fsType: ext4
```
Comment (bgrant0607) [inline examples/gce-pd/v1beta3/testpd.yaml]: Needs newline

File: examples/gce-pd/v1beta3/testpd.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+apiVersion: v1beta3
+kind: Pod
+metadata:
+  labels:
+    test: testpd
+  name: testpd
+spec:
+  containers:
+    - image: kubernetes/pause
+      name: testpd
```
Comment (bgrant0607) [inline examples/gce-pd/v1beta3/testpd.yaml]: Please put the name first.

File: examples/redis/v1beta3/redis-sentinel-controller.yaml (hunk 1 above)
Comment (bgrant0607) [inline examples/redis/v1beta3/redis-sentinel-controller.yaml]: Yes, please remove the name.

File: pkg/api/v1beta3/types.go [hunk 5]
```diff
@@ -379,6 +379,8 @@ type Container struct {
 	// Optional: Default to false.
 	Privileged bool `json:"privileged,omitempty"`
 	// Optional: Policy for pulling images for this container
+    // Defaults to PullAlways if the image tag is latest,
```
Comment (bgrant0607) [inline pkg/api/v1beta3/types.go]: Please revert. @vishh could do this with the addition of descriptions, though the descriptions are most important.

File: examples/redis/v1beta3/redis-proxy.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1beta3
```
Comment (bgrant0607) [inline examples/redis/v1beta3/redis-proxy.yaml]: This should really be a one-replica replicationController.

File: pkg/api/v1beta3/types.go (hunk 5 above)
Comment (vishh) [inline pkg/api/v1beta3/types.go]: #5070 should take care of this.

File: examples/redis/v1beta3/redis-proxy.yaml (hunk 6 above)
Comment (ArtfulCoder) [inline examples/redis/v1beta3/redis-proxy.yaml]: Fixed.

**Comment by ArtfulCoder** [comment]
PTAL @bgrant0607

File: examples/redis/v1beta3/redis-slave.yaml [hunk 7]
```diff
@@ -0,0 +1,21 @@
+apiVersion: v1beta3
+kind: Pod
```
Comment (bgrant0607) [inline examples/redis/v1beta3/redis-slave.yaml]: We should delete this file. They should be using the replication controller now.

File: examples/redis/v1beta3/redis-sentinel.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
+apiVersion: v1beta3
+kind: Pod
```
Comment (bgrant0607) [inline examples/redis/v1beta3/redis-sentinel.yaml]: We should delete this file. They should be using the replication controller now.

File: examples/redis/v1beta3/redis-master.yaml [hunk 9]
```diff
@@ -0,0 +1,33 @@
+apiVersion: v1beta3
+kind: Pod
```
Comment (bgrant0607) [inline examples/redis/v1beta3/redis-master.yaml]: We should delete this file. They should be using the replication controller now.

File: examples/redis/v1beta3/README.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+
+### Conclusion
+At this point we now have a reliable, scalable Redis installation.  By resizing the replication controller for redis servers, we can increase or decrease the number of read-slaves in our cluster.  Likewise, if failures occur, the redis-sentinels will perform master election and select a new master. 
+
+### tl; dr
+For those of you who are impatient, here is the summary of commands we ran in this tutorial
+
+```sh
+# Create a bootstrap master
+kubectl create -f examples/redis/v1beta3/redis-master.yaml
```
Comment (bgrant0607) [inline examples/redis/v1beta3/README.md]: Please ensure that all these commands are creating replication controllers, not raw pods.

File: examples/redis/v1beta3/README.md [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+This will create two additional replicas of the redis server and two additional replicas of the redis sentinel.
+
+Unlike our original redis-master pod, these pods exist independently, and they use the ```redis-sentinel-service``` that we defined above to discover and join the cluster.
+
+### Delete our manual pod
+The final step in the cluster turn up is to delete the original redis-master pod that we created manually.  While it was useful for bootstrapping discovery in the cluster, we really don't want the lifespan of our sentinel to be tied to the lifespan of one of our redis servers, and now that we have a successful, replicated redis sentinel service up and running, the binding is unnecessary.
+
+Delete the master as follows:
+```sh
+kubectl delete pods redis-master
```
Comment (bgrant0607) [inline examples/redis/v1beta3/README.md]: They should use `kubectl stop rc redis-master`

File: examples/redis/v1beta3/README.md [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+is a [_Pod_](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/pods.md).  A Pod is one or more containers that _must_ be scheduled onto the same host.  All containers in a pod share a network namespace, and may optionally share mounted volumes.
+
+We will used the shared network namespace to bootstrap our Redis cluster.  In particular, the very first sentinel needs to know how to find the master (subsequent sentinels just ask the first sentinel).  Because all containers in a Pod share a network namespace, the sentinel can simply look at ```$(hostname -i):6379```.
+
+Here is the config for the initial master and sentinel pod:
+[redis-master](redis-master.yaml)
+
+Create this master as follows:
+```sh
+kubectl create -f examples/redis/v1beta3/redis-master.yaml
```
Comment (bgrant0607) [inline examples/redis/v1beta3/README.md]: Please ensure that all these commands are creating replication controllers, not raw pods.

**Comment by ArtfulCoder** [comment]
All pods are created through replication controllers now.
Updated the command to use "kubectl stop rc redis-master".

PTAL @bgrant0607

**Comment by bgrant0607** [comment]
@ArtfulCoder Did you decide that it was ok to convert all the pods to replication controllers?

**Comment by ArtfulCoder** [comment]
@bgrant0607  PTAL at the changes.

**Comment by bgrant0607** [comment]
Great! LGTM.

**Comment by bgrant0607** [comment]
You deleted a file used outside the example:
```
valid-pod
F0311 21:40:40.301610   19097 create.go:50] Client error processing command: the path "examples/redis/redis-proxy.yaml" does not exist
!!! Error in ./hack/test-cmd.sh:240
  'kubectl create -f examples/redis/redis-proxy.yaml "${kube_flags[@]}"' exited with status 255
Call stack:
  1: ./hack/test-cmd.sh:240 main(...)
```

**Comment by ArtfulCoder** [comment]
@bgrant0607 PTAL
All checks are passing now.

**Pull request merged**